### DISCUSSION TITLE
Thoughts on using `get_icon_color` in tabline

### DISCUSSION THREAD

**Discussion by richin13** [nvim-mini/mini.nvim#799] [Ideas]
I'm in the process of migrating from heirline to mini.{tabline && statusline}. So far I'm enjoying the transition except for the lack of colors here and there. I was wondering if it would be possible to add colors to the file icons in the tabline by using `get_icon_color` from devicons (challenge being it returns a hex code instead of a highlight group name)

![image](https://github.com/echasnovski/mini.nvim/assets/8370058/696a5044-44b2-49f8-9c2d-8d336988f2a4)

![image](https://github.com/echasnovski/mini.nvim/assets/8370058/313030aa-8bcc-4239-84e1-2757f232c5a9)

I can live without the other customizations but the lack of colors here is throwing me off a bit (😆)

So far I've taken it pretty close with this patch:

```diff
diff --git a/lua/mini/tabline.lua b/lua/mini/tabline.lua
index a7d879b..fba873a 100644
--- a/lua/mini/tabline.lua
+++ b/lua/mini/tabline.lua
@@ -364,10 +364,13 @@ H.finalize_labels = function()
   for _, tab in pairs(H.tabs) do
     if show_icons and has_devicons then
       local extension = vim.fn.fnamemodify(tab.label, ':e')
-      local icon = devicons.get_icon(tab.label, extension, { default = true })
-      tab.label = string.format(' %s %s ', icon, tab.label)
+      local icon, color = devicons.get_icon_color(tab.label, extension, { default = true })
+      local hl = "CustomIconColor"..extension
+      vim.api.nvim_set_hl(0, hl, { fg = color })
+      tab.icon = "%#"..hl.."#" .. string.format(' %s ', icon)
     else
       tab.label = string.format(' %s ', tab.label)
+      tab.icon = ""
     end
   end
 end
@@ -472,7 +475,7 @@ H.concat_tabs = function()
   local t = {}
   for _, tab in ipairs(H.tabs) do
     -- Escape '%' in labels
-    table.insert(t, ('%s%s%s'):format(tab.hl, tab.tabfunc, tab.label:gsub('%%', '%%%%')))
+    table.insert(t, ('%s%s%s%s'):format(tab.icon, tab.hl, tab.tabfunc, tab.label:gsub('%%', '%%%%')))
   end
 
   -- Usage of `%X` makes filled space to the right 'non-clickable'
```

But there are some issues (such as the icon not being clickable and the highlight being weird (the bg mostly)) but I wanted to survey the interest in something like this or if it has been discussed/considered in the past but discarded for other reasons/philosophies before I continue down this rabbit hole (not too knowledgeable on lua tbh)

![image](https://github.com/echasnovski/mini.nvim/assets/8370058/9983dc42-b16c-414d-b85f-0d8e836a60db)

**Comment by echasnovski** [comment]
Thanks for your interest in 'mini.tabline'!

Yes, it was suggested earlier in #153. The resolution is in [this comment](https://github.com/echasnovski/mini.nvim/issues/153#issuecomment-1445121506) with one of the points being similar to what you describe about highlighting colors.

I *might* revisit this soon after some new planned 'mini.nvim' modules come out which *might* make this issue more solvable.

**Reply by richin13** [reply]
Understandable, thank you very much!
